### DISCUSSION TITLE
Position arrow orientation vs. map rotation

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#15570] [Q&A]
I just wanted to share my observation and understand the codewise background for this:

While on bike tour intensively using the rotating offline maps (with unified map) I came across situations where the rotation did not match my direction. Of course there are numerous factors involved here...most likely my device sensors are the culprit here causing some offset or just delivering wrong data.

Anyway I noticed the following:
I would have expected, that when using the map in rotation mode the position arrow should always point upwards. However the methods for rotating the position arrow and for rotating the map seem to be different in some way.
There have been several situations, where the arrow was not pointing upwards on the map. In some cases the arrow was performing better (in terms of representing my moving direction) and in some cases the map rotation was performing better.

I have used the precise mode throughout, so it should not be caused by delay in rotation.

I would have expected, that map rotation just "freezes" the arrow upwards and instead rotate the map around it, while on northbound 
mode the map is "frozen" and the arrow rotating using the same method.

Can someone try to explain what the difference between rotation of the arrow and rotation of the map is?

Once again:
I don't think we should change anything here yet, just first of all I am trying to understand and take some more observation to see how to improve rotation accuracy (if possible).

**Comment by fm-sys** [comment]
IIRC someone introduced that even the precise mode will switch to low power (or something in between) if the speed is larger than 5 kmh.

I totally agree/believe that precise mode should always be precise. Otherwise it's useless and looks broken IMHO. 

(the real precise mode will always be in sync with the heading indicator)

**Comment by moving-bits** [comment]
I have had similar experiences in the past, but not really reproducible. Happened for me in both "low power" and "precise" modes - blue arrow got stuck at a certain position, maybe something like 15°-20° off. After a while it fixed itself, not sure anymore if any action from me was involved (like closing and reopening map and/or c:geo).

**Reply by moving-bits** [reply]
Used the nice weather today for a few kilometer's walk. testing all five autorotation variants ("old" GM with low-power autorotation, and both UnifiedMap variants with both autorotation variants). Overall, the result were pretty mixed today:

- There are differences that are to be expected by the differences between the two modes:
  - "low power" mode: rotation can be lagging, as it has a minimum difference before it rotates
  - "precise" mode: map is pretty shaky sometimes, nearly constantly rotating a few degrees to the left and back to the right

Besides that, the user experience in all modes was pretty similar, the artifacts not identical, but seemingly more depending on current signal reception as on map type and rotation mode:
- I had parts along my trip where the blue indicator was pointing exactly into direction of walking, but the map was orientated about 90° off. This persisted for a couple of minutes, and came back later during my walk (so this was not due to GPS not yet being initialized).
- but I had also parts where both was (mostly) fitting, blue arrow pointing into direction of walking, and map below in accordance to the actual landscape
- what I did not experience today, but remember from earlier trips, is the map being oriented exactly according to my direction of travel, but the blue position marker being off by a constant degree

Some additional info:
- GPS was already running for quite some time before I started the walk. Position was on spot, accuracy circle minimal.
- I was holding my mobile horizontally throughout all times when I was taking readings. From time to time I rotated it a bit to see if and when the map adjusted - behavior for this was mostly as expected (quick if in "precise" mode, delayed in "low power" mode - but for both modes not working always)

I cannot say currently what the causes for this behavior are. Some ideas:
- sensor problems in my mobile (Samsung S22 in this case)
- problems with GPS signal
- some misinterpretation of heading/bearing or some other implementation quirk on our side
- ...

(Having said that, it's not as bad as it sounds. Orientation using our map was always possible.)
